### PR TITLE
[Fix #3202] Register proper offense in `Style/EmptyElse`

### LINKED ISSUE #3202: Style/EmptyElse crash on master
source method:
```
def return_time
    if round_trip
      last_trip.scheduled_start_at
    else
      nil
    end
end
```

for `rubocop`:
```
undefined method `begin_pos' for #<Parser::Source::Map::Condition:0x007f971cdff970>
gems/rubocop-9cf9c3c91446/lib/rubocop/cached_data.rb:39:in `serialize_offense'
gems/rubocop-9cf9c3c91446/lib/rubocop/cached_data.rb:18:in `block in to_json'
gems/rubocop-9cf9c3c91446/lib/rubocop/cached_data.rb:18:in `map'
gems/rubocop-9cf9c3c91446/lib/rubocop/cached_data.rb:18:in `to_json'
gems/rubocop-9cf9c3c91446/lib/rubocop/result_cache.rb:93:in `block in save'
gems/rubocop-9cf9c3c91446/lib/rubocop/result_cache.rb:92:in `open'
gems/rubocop-9cf9c3c91446/lib/rubocop/result_cache.rb:92:in `save'
gems/rubocop-9cf9c3c91446/lib/rubocop/runner.rb:145:in `save_in_cache'
gems/rubocop-9cf9c3c91446/lib/rubocop/runner.rb:89:in `process_file'
gems/rubocop-9cf9c3c91446/lib/rubocop/runner.rb:59:in `block in inspect_files'
gems/rubocop-9cf9c3c91446/lib/rubocop/runner.rb:57:in `each'
gems/rubocop-9cf9c3c91446/lib/rubocop/runner.rb:57:in `inspect_files'
gems/rubocop-9cf9c3c91446/lib/rubocop/runner.rb:35:in `run'
```
for `rubocop -C false`:
```
C: Style/EmptyElse: Redundant else-clause.
undefined method `source_line' for #<Parser::Source::Map::Condition:0x007f7f90dd6eb0>
gems/rubocop-9cf9c3c91446/lib/rubocop/formatter/clang_style_formatter.rb:36:in `valid_line?'
gems/rubocop-9cf9c3c91446/lib/rubocop/formatter/clang_style_formatter.rb:25:in `report_offense'
gems/rubocop-9cf9c3c91446/lib/rubocop/formatter/clang_style_formatter.rb:13:in `block in report_file'
gems/rubocop-9cf9c3c91446/lib/rubocop/formatter/clang_style_formatter.rb:13:in `each'
gems/rubocop-9cf9c3c91446/lib/rubocop/formatter/clang_style_formatter.rb:13:in `report_file'
gems/rubocop-9cf9c3c91446/lib/rubocop/formatter/progress_formatter.rb:43:in `block in finished'
gems/rubocop-9cf9c3c91446/lib/rubocop/formatter/progress_formatter.rb:42:in `each'
gems/rubocop-9cf9c3c91446/lib/rubocop/formatter/progress_formatter.rb:42:in `finished'
gems/rubocop-9cf9c3c91446/lib/rubocop/formatter/formatter_set.rb:30:in `block (3 levels) in <class:FormatterSet>'
gems/rubocop-9cf9c3c91446/lib/rubocop/formatter/formatter_set.rb:30:in `each'
gems/rubocop-9cf9c3c91446/lib/rubocop/formatter/formatter_set.rb:30:in `block (2 levels) in <class:FormatterSet>'
gems/rubocop-9cf9c3c91446/lib/rubocop/runner.rb:68:in `inspect_files'
gems/rubocop-9cf9c3c91446/lib/rubocop/runner.rb:35:in `run'
```

--------

## Expected behavior
No crash
## Actual behavior
Crash
## Steps to reproduce the problem
run rubocop on test.rb:
```
def truish
  if true
    true
  else
    nil
  end
end
```

## RuboCop version
0.40.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-darwin15)

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
Fixes #3202. Unfortunately I couldn't find a better way for a regression test than this one... :(

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it)
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

File: spec/rubocop/cop/style/empty_else_spec.rb [hunk 1]
```diff
@@ -35,6 +35,18 @@
...
 
+  shared_examples_for 'offense registration' do
+    it 'registers an offense with correct message' do
+      inspect_source(cop, source)
+      expect(cop.messages).to eq(['Redundant `else`-clause.'])
+    end
+
+    it 'registers an offense with correct location' do
+      inspect_source(cop, source)
+      expect(cop.offenses.first.location).to be_a(Parser::Source::Range)
```
Comment (jonas054) [inline spec/rubocop/cop/style/empty_else_spec.rb]: Checking the type is a bit too whitebox IMO. I would prefer 
```ruby
expect(cop.highlights).to eq(['else'])
```
That would also catch the bug.

**Comment by jonas054** [comment]
Nice job on the refactoring. I just had one comment on the specs.

File: spec/rubocop/cop/style/empty_else_spec.rb (hunk 1 above)
Comment (deivid-rodriguez) [inline spec/rubocop/cop/style/empty_else_spec.rb]: Absolutely, that type check was bothering me too. Updated!

**Comment by jonas054** [comment]
:+1: I'm satisfied.

**Comment by bbatsov** [comment]
This has to be rebased.

**Comment by bbatsov** [comment]
Btw, I'd reword your second commit message to be exactly what you used for the PR description itself.

**Comment by deivid-rodriguez** [comment]
Rebased and changed commit message
